src/SourceWatcherFuncApp/DiagSourceWatcher.cs

```diff
@@ -0,0 +1,87 @@
+        }
+
+        [FunctionName("DiagSourceWatcher")]
+        public async Task Run([TimerTrigger("0 0 0 * * *")]TimerInfo timerInfo, ILogger log)
+        {
+            log.LogInformation($"C# Timer trigger function executed at: {DateTime.Now}");
+
+            log.LogInformation($"Loading directories from github branch  {config["Github:Branch"]}");
+
+            var githubDirectories = await githubService.DownloadGithubDirectories(config["Github:Branch"]);
+          
+            foreach (var githubdir in githubDirectories)
+            {
+
+                if (!githubdir.Type.Equals("dir", StringComparison.OrdinalIgnoreCase)) continue;
+                
+                var contentList = await githubService.DownloadGithubDirectories(branchdownloadUrl: githubdir.Url);
+                
+                var assemblyFile = contentList.Where(githubFile => githubFile.Name.EndsWith("dll")).FirstOrDefault();
+                var scriptFile = contentList.Where(githubfile => githubfile.Name.EndsWith(".csx")).FirstOrDefault();
+                var configFile = contentList.Where(githubFile => githubFile.Name.Equals("package.json", StringComparison.CurrentCultureIgnoreCase)).FirstOrDefault();
+
+                if (assemblyFile != null && scriptFile != null && configFile != null)
+                {
+
+                    log.LogInformation($"Getting content for Assembly file : {assemblyFile.Path}");
+                    var assemblyData = await githubService.GetFileContentStream(assemblyFile.Download_url);
+
+                    log.LogInformation("Reading detector metadata");
+                    var configFileData = await githubService.GetFileContentByType<DetectorEntity>(configFile.Download_url);
+                    var scriptData = await githubService.GetFileContentByType<string>(scriptFile.Download_url);
+
+                    configFileData = EntityHelper.PrepareEntityForLoad(assemblyData, scriptData, configFileData);
+                    configFileData.GitHubSha = githubdir.Sha;
+                    configFileData.GithubLastModified = await githubService.GetCommitDate(scriptFile.Path);
+                    
+                    // First check if table already has the entry
+                    var existingDetectorEntity = await storageService.GetEntityFromTable(configFileData.PartitionKey, configFileData.RowKey);
+
+                    // If there is no entry in table or github last modifed date has been changed, upload to blob
+                    if (existingDetectorEntity == null || existingDetectorEntity.GithubLastModified != configFileData.GithubLastModified)
+                    {                        
+                        var assemblyLastModified = await githubService.GetCommitDate(assemblyFile.Path);
+                        blobService.LoadBlobToContainer(assemblyFile.Path, assemblyData);
+                    }
+
+                    var result = await storageService.LoadDataToTable(configFileData);
```

Comment:
You only need to do this within the above if body right?

Reply:
Nope. This should be done in case of any of changes to the detector entity. For example, if `supportTopicList `is updated or `InternalOnly `flag is updated in the publish flow. Instead of comparing and checking which fields of `DetectorEntity` has changed and updating those, I am just passing the model so that table storage can handle that better